core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -18,39 +18,35 @@
  */
 
 import { downgradeInjectable } from '@angular/upgrade/static';
-import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 
-import { ContextService } from 'services/context.service';
-import { UrlInterpolationService } from
-  'domain/utilities/url-interpolation.service';
+import { ExplorationPermissionsBackendApiService } from
+  'domain/exploration/exploration-permissions-backend-api.service';
+import { ExplorationPermissions } from
+  'domain/exploration/ExplorationPermissionsObjectFactory';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserExplorationPermissionsService {
   constructor(
-    private contextService: ContextService,
-    private http: HttpClient,
-    private urlInterpolationService: UrlInterpolationService) {}
+    private explorationPermissionsBackendApiService:
+    ExplorationPermissionsBackendApiService) {}
 
-  static permissionsPromise: Promise<object> = null;
-  getPermissionsAsync(): Promise<object> {
-    if (!UserExplorationPermissionsService.permissionsPromise) {
-      let explorationPermissionsUrl = this.urlInterpolationService
-        .interpolateUrl( '/createhandler/permissions/<exploration_id>', {
-          exploration_id: this.contextService.getExplorationId()
-        });
+  static permissionsPromise: Promise<ExplorationPermissions> = null;
 
+  getPermissionsAsync(): Promise<ExplorationPermissions> {
+    if (!UserExplorationPermissionsService.permissionsPromise) {
       UserExplorationPermissionsService.permissionsPromise = (
-        this.http.get(explorationPermissionsUrl).toPromise().then(
-          (response) => {
-            return response;
+        this.explorationPermissionsBackendApiService.getPermissions().then(
+          (permissions) => {
+            return permissions;
           }
         ));
```

Comment:
```suggestion
          permissions => permissions));
```

Reply:
Actually, I'm confused with this. `UserExplorationPermissionsService.permissionsPromise` expects a promise, shouldn't it just return `this.explorationPermissionsBackendApiService.getPermissions()`?

Reply:
Yeah, right. done